docker/Dockerfile.test-env

```diff
@@ -144,10 +145,14 @@ RUN if [ "$MPI" = "mpich" ]; then \
     ldconfig && \
     rm -rf /tmp/*
 
+# Create virtual env
```

Comment:
To avoid having to pass `--break-system-packages`, an even simpler trick that doesn't require creating a venv is to `rm /usr/lib/python3.12/EXTERNALLY-MANAGED`